arrow/parser.py

```diff
@@ -275,15 +275,27 @@ def _build_datetime(parts):
         elif am_pm == "am" and hour == 12:
             hour = 0
 
-        return datetime(
-            year=parts.get("year", 1),
-            month=parts.get("month", 1),
-            day=parts.get("day", 1),
-            hour=hour,
-            minute=parts.get("minute", 0),
-            second=parts.get("second", 0),
-            microsecond=parts.get("microsecond", 0),
-            tzinfo=parts.get("tzinfo"),
+        microsecond = parts.get("microsecond", 0)
```

Comment:
Let's add a comment here to denote that we are accounting for the rounding up to `1000000` so that future contributors are aware of the new functionality :).